src/c/c_client.cpp

```diff
@@ -30,517 +30,654 @@
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string name_str(name, name_length);
+  std::string new_name_str(new_name, new_name_length);
+
   s->rename_dataset(name_str, new_name_str);
-  return;
 }
 
+// Copy a dataset from the src_name to the dest_name
 extern "C"
-void copy_dataset(void* c_client, const char* src_name,
-                  const size_t src_name_length, const char* dest_name,
-                  const size_t dest_name_length
-                  )
+void copy_dataset(void *c_client, const char *src_name,
+                  const size_t src_name_length, const char *dest_name,
+                  const size_t dest_name_length)
 {
-  /* Copy a dataset from teh src_name to the dest_name
-  */
-  Client* s = (Client *)c_client;
-  std::string src_name_str = std::string(src_name, src_name_length);
-  std::string dest_name_str = std::string(dest_name, dest_name_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == src_name || NULL == dest_name)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string src_name_str(src_name, src_name_length);
+  std::string dest_name_str(dest_name, dest_name_length);
+
   s->copy_dataset(src_name_str, dest_name_str);
-  return;
 }
 
+// Delete a dataset (all metadata and tensors) from the database.
 extern "C"
-void delete_dataset(void* c_client, const char* name,
-                    const size_t name_length)
+void delete_dataset(void *c_client, const char *name, const size_t name_length)
 {
-  /* Delete a dataset (all metadata and tensors) from the
-  database.
-  */
-  Client* s = (Client *)c_client;
-  std::string dataset_name = std::string(name, name_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == name)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string dataset_name(name, name_length);
   s->delete_dataset(dataset_name);
-  return;
 }
 
+// Put a tensor of a specified type into the database
 extern "C"
-void put_tensor(void* c_client,
-                const char* key,
+void put_tensor(void *c_client,
+                const char *key,
                 const size_t key_length,
-                void* data,
-                const size_t* dims,
+                void *data,
+                const size_t *dims,
                 const size_t n_dims,
-                CTensorType type,
-                CMemoryLayout mem_layout)
+                const CTensorType type,
+                const CMemoryLayout mem_layout)
 {
-  /* Put a tensor of a specified type into the database
-  */
-  Client* s = (Client *)c_client;
-  std::string key_str = std::string(key, key_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == key || NULL == data || NULL == dims)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
```

Comment:
Thinking more about it, I think it would be more helpful to have this kind of behavior documented ahead of the function (in the header file) instead of in the function.

Reply:
Not clear what you're asking for here

Reply:
Sorry this wasn't clear.  Where appropriate I was thinking it would be helpful to have this temporary error handling documented (no action error handling or returning NULL pointer, etc) in the doc strings in the header files.  But since this is temporary, it's not a big deal at all.

Reply:
Done